Convert XML entities

emg36 opened the pull request:
Replace potential XML valid symbols that intefer with rendering a pdf

emg36 committed:
Convert XML entities

Replace potential XML valid symbols that intefer with rendering a pdf

emg36 committed:
Python versions

suspectpart commented:
I stumbled on the exact same problem, see #49.

Could we talk about having this as a feature and revise this solution? If we dropped Python 2 support (it's about time!), we could just use the [html.escape](https://docs.python.org/3/library/html.html#html.escape) standard lib function. Or, if Python 2 compatibility is still desired, run with the solution by @emg36 as it is exactly doing what `html.escape` [is doing anyway](https://github.com/python/cpython/blob/3.9/Lib/html/__init__.py#L12).

suspectpart commented:
@revolunet Are you still maintaining this library?

emg36 committed:
remove six and include escaping

- use html escaping instead of .replace
- a conditional on the pdf data to skip keys with and underscore and/or have no key. This was to remove dictionaries that were being passed in, not just a key string value pair

Commit:
Merge branch 'master' of https://github.com/revolunet/pypdftk into develop

# Conflicts:
#	pypdftk.py

emg36 committed:
Merge pull request #1 from emg36/develop

merge in original project
